File: package/ios/ClusterBadgeMetrics.swift
import CoreGraphics

/// Shared cluster badge sizing used by merge tests and drawn annotation views.
enum ClusterBadgeMetrics {
  static let mergeGap: Double = 6

  static func badgeRadius(for count: Int) -> Double {
    switch count {
    case ..<2:
      return 14
    case ..<10:
      return 17
    case ..<100:
      return 20
    case ..<1000:
      return 24
    default:
      return 28
    }
  }

  static func diameter(for count: Int) -> CGFloat {
    switch count {
    case ..<10:
      return 34
    case ..<100:
      return 40
    case ..<1000:
      return 48
    default:
      return 56
    }
  }
}

File: package/android/src/main/java/com/margelo/nitro/nitromaps/ClusterBadgeMetrics.kt
package com.margelo.nitro.nitromaps

/** Shared cluster badge sizing used by merge tests and drawn icons. */
internal object ClusterBadgeMetrics {
  const val MERGE_GAP_DP = 6.0

  fun badgeRadiusDp(count: Int): Double = when {
    count < 2 -> 14.0
    count < 10 -> 17.0
    count < 100 -> 20.0
    count < 1000 -> 24.0
    else -> 28.0
  }

  fun diameterDp(count: Int): Float = when {
    count < 10 -> 34f
    count < 100 -> 40f
    count < 1000 -> 48f
    else -> 56f
  }
}
